tests/constraints.test/t12.req

```diff
@@ -41,3 +44,40 @@ select * from p order by i
 
 exec procedure sys.cmd.verify("p")
 exec procedure sys.cmd.verify("c")
+
+
+select "test #4" as comment
+delete from c;
+delete from p;
+insert into p values (1)
+insert into c values (1)
+# this is same as test 1
+begin
+update c set i = 2 where i = 1
+update p set i = 2 where i = 1
+commit
+select * from c order by i
+select * from p order by i
+
+exec procedure sys.cmd.verify("p")
+exec procedure sys.cmd.verify("c")
+
+
+select "test #5" as comment
+delete from c;
+delete from p;
+insert into p values (1)
+insert into p values (3)
+insert into c values (1)
+# this is same as test 1
+begin
+update c set i = 3 where i = 1
+update p set i = 2 where i = 1
```

Comment:
I am a bit confused here. Shouldn't this fail given the record (i==1) would not be found in child during cascade?

Reply:
So the cascade has no work to do on child tbl because txn already updated child entry i==1 to 3.